### PR TITLE
Feature: Study Rooms

### PR DESCRIPTION
**Pull request by brdvlps**
Finally, the long-awaited study rooms feature. See #231.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/activities/generic/ActivityForSearching.java [hunk 1]
```diff
@@ -92,7 +92,6 @@ public boolean onCreateOptionsMenu(Menu menu) {
         // If activity gets called via Intent with a search query set SearchView accordingly
         if (mQuery != null) {
             mSearchView.setQuery(mQuery, false);
-            mOpenSearch = true;
```
Comment (brdvlps) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/generic/ActivityForSearching.java]: Notice I changed the behavior here, i.e. the search textbox won't be opened if there is a predefined query.
Makes more sense imho.

**Comment by kordianbruck** [comment]
Awesome! Please fix the lint errors so we can proceed to merge 👍

**Comment by brdvlps** [comment]
Done. I ignored the remaining errors concerning the new 24.0.0 alpha of the android support libraries.

**Comment by pfent** [comment]
@brdvlps thanks again for the PR, I finally came around to merge it...

Please note, that I changed the generic base Activity, since I got some loading problems, when my phone initially was only connected via mobile data. See: dcfa28ba6b5b2c120503ef86b924117037ecddc7

**Pull request merged**